src/ophyd_async/epics/areadetector/controllers/ad_controller.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+from typing import Optional
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    DetectorControl,
+    DetectorTrigger,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+
+from ..drivers.ad_driver import ADDriver, ImageMode
+
+
+class ADController(DetectorControl):
+    def __init__(self, driver: ADDriver) -> None:
+        self.driver = driver
+
+    def get_deadtime(self, exposure: float) -> float:
+        return 0.002
+
+    async def arm(
+        self,
+        trigger: DetectorTrigger = DetectorTrigger.internal,
+        num: int = 0,
+        exposure: Optional[float] = None,
+    ) -> AsyncStatus:
+        frame_timeout = DEFAULT_TIMEOUT + await self.driver.acquire_time.get_value()
+        await asyncio.gather(
+            self.driver.num_images.set(num),
+            self.driver.image_mode.set(ImageMode.single),
+        )
+        return await set_and_wait_for_value(
+            self.driver.acquire, True, timeout=frame_timeout
+        )
+
+    async def disarm(self):
+        # wait=False means don't caput callback. We can't use caput callback as we
+        # already used it in arm() and we can't have 2 or they will deadlock
+        await self.driver.acquire.set(False, wait=False)
+        await wait_for_value(self.driver.acquire, False, timeout=1)
```

Comment:
These 2 lines are likely to be used in many areaDetector `ADControl` subclasses, should they go in a utility function with a suitable name?

Reply:
Could I modify `set_and_wait_for_value` to take an optional `wait` kwarg, so that we can replace these two with a `set_and_wait_for_value call`?

Reply:
That would make sense from a code point of view, but I worry about writing `await set_and_wait_for_value(self.drv.acquire, False, wait=False)`. 3 waits in the same line makes my brain hurt! I'd prefer a separate function so we could write `await stop_busy_record(self.drv.acquire)` which makes more sense for an EPICS person

Reply:
Sure, I can understand this. How about,

`await set_and_wait_for_value(self.drv.acquire, False, with_callback=False)`?

Reply:
resolution; I've made a new method in area detector utils, stop_busy_record, which does essentially this.